ares/fc/system/system.cpp

```diff
@@ -109,6 +109,14 @@ auto System::power(bool reset) -> void {
   apu.power(reset);
   ppu.power(reset);
   scheduler.power(cpu);
+
+  // The CPU takes 8 cycles before it starts executing
+  // the ROM's code after a reset/power up
+  for (auto i : range(8)) {
```

Comment:
This part looks rather suspect to me; since it is calling main functions of other portions out of band as well as accessing cpu.thread directly.


Consider updating CPU::power() in ares/fc/cpu/cpu.cpp to just call step(rate() *8 ); instead: I will work on implementing the proper 6502 reset process in a later commit, but this change should be enough to allow us to merge.